modules/shared/src/main/scala/org/tessellation/currency/schema/FeeTransaction.scala

```diff
@@ -0,0 +1,22 @@
+package org.tessellation.currency.schema
+
+import org.tessellation.ext.derevo.ordering
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+import org.tessellation.schema.transaction.{TransactionOrdinal, TransactionReference, TransactionSalt}
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+
+@derive(decoder, encoder, order, ordering, show)
+case class FeeTransaction(
+  source: Address,
+  destination: Address,
+  amount: Amount,
+  parent: TransactionReference,
```

Comment:
Hmm, according to ticket "It must have a unique FeeTransactionReference distinct from other transaction references". Are we staying with `TransactionReference` anyway?

Reply:
Yeah we talked about it and it seems it's not strictly necessary to not reuse `TransactionReference` schema itself as long as we set the values correctly in the code - e.i. we don't use regular tx refs as fee tx refs by mistake. We could use a new `FeeTransactionReference` schema if we decide that this will help us in not making some programmatic error.

Reply:
I support that. We have many places in the code where we operate on references, so it would be easier to see a different type

Reply:
Cool let's do that then.